Compling with musl libc (Alpine Linux)

Compiling with alpine/[musl libc](https://musl.libc.org/), which is a common libc used in Python & Docker environments, does not support exception management `fegetexcept`, `feenableexcept`, `fedisableexcept`. They seem to be GNU-only.
- https://man7.org/linux/man-pages/man3/feenableexcept.3.html
- https://linux.die.net/man/3/fedisableexcept

We need to guard the respective functions in `AMReX.cpp` with the GNU libc macro `HAVE_FEDISABLEEXCEPT` (double check name, we can also use other checks to make sure GNU libc is used/not used).

Error message:
```

2025-03-12T18:45:57.8097333Z /project/amrex/Src/Base/AMReX.cpp: In function 'amrex::AMReX* amrex::Initialize(int&, char**&, bool, MPI_Comm, const std::function<void()>&, std::ostream&, std::ostream&, ErrorHandler)':
2025-03-12T18:45:57.8099222Z /project/amrex/Src/Base/AMReX.cpp:592:36: error: 'fegetexcept' was not declared in this scope; did you mean 'fetestexcept'?
2025-03-12T18:45:57.8101686Z   592 |                 prev_fpe_excepts = fegetexcept();
2025-03-12T18:45:57.8102198Z       |                                    ^~~~~~~~~~~
2025-03-12T18:45:57.8102649Z       |                                    fetestexcept
2025-03-12T18:45:57.8144630Z /project/amrex/Src/Base/AMReX.cpp:594:21: error: 'feenableexcept' was not declared in this scope; did you mean 'feraiseexcept'?
2025-03-12T18:45:57.8146351Z   594 |                     feenableexcept(curr_fpe_excepts);  // trap floating point exceptions
2025-03-12T18:45:57.8147659Z       |                     ^~~~~~~~~~~~~~
2025-03-12T18:45:57.8148211Z       |                     feraiseexcept
2025-03-12T18:45:57.8241552Z /project/amrex/Src/Base/AMReX.cpp: In function 'void amrex::Finalize(AMReX*)':
2025-03-12T18:45:57.8242876Z /project/amrex/Src/Base/AMReX.cpp:830:13: error: 'fedisableexcept' was not declared in this scope; did you mean 'feraiseexcept'?
2025-03-12T18:45:57.8244687Z   830 |             fedisableexcept(curr_fpe_excepts);
2025-03-12T18:45:57.8245754Z       |             ^~~~~~~~~~~~~~~
2025-03-12T18:45:57.8246205Z       |             feraiseexcept
2025-03-12T18:45:57.8286837Z /project/amrex/Src/Base/AMReX.cpp:831:13: error: 'feenableexcept' was not declared in this scope; did you mean 'feraiseexcept'?
2025-03-12T18:45:57.8287996Z   831 |             feenableexcept(prev_fpe_excepts);
2025-03-12T18:45:57.8288608Z       |             ^~~~~~~~~~~~~~
2025-03-12T18:45:57.8290043Z       |             feraiseexcept
2025-03-12T18:45:57.8384040Z /project/amrex/Src/Base/AMReX.cpp: In function 'amrex::FPExcept amrex::getFPExcept()':
2025-03-12T18:45:57.8385386Z /project/amrex/Src/Base/AMReX.cpp:946:20: error: 'fegetexcept' was not declared in this scope; did you mean 'fetestexcept'?
2025-03-12T18:45:57.8387144Z   946 |     auto excepts = fegetexcept();
2025-03-12T18:45:57.8387583Z       |                    ^~~~~~~~~~~
2025-03-12T18:45:57.8387965Z       |                    fetestexcept
2025-03-12T18:45:57.8440571Z /project/amrex/Src/Base/AMReX.cpp: In function 'amrex::FPExcept amrex::setFPExcept(FPExcept)':
2025-03-12T18:45:57.8441877Z /project/amrex/Src/Base/AMReX.cpp:959:5: error: 'fedisableexcept' was not declared in this scope; did you mean 'disableFPExcept'?
2025-03-12T18:45:57.8443618Z   959 |     fedisableexcept(flags);
2025-03-12T18:45:57.8444166Z       |     ^~~~~~~~~~~~~~~
2025-03-12T18:45:57.8444686Z       |     disableFPExcept
2025-03-12T18:45:57.8495493Z /project/amrex/Src/Base/AMReX.cpp:964:5: error: 'feenableexcept' was not declared in this scope; did you mean 'feraiseexcept'?
2025-03-12T18:45:57.8514793Z   964 |     feenableexcept(flags);
2025-03-12T18:45:57.8515706Z       |     ^~~~~~~~~~~~~~
2025-03-12T18:45:57.8516209Z       |     feraiseexcept
2025-03-12T18:45:57.8552656Z /project/amrex/Src/Base/AMReX.cpp: In function 'amrex::FPExcept amrex::disableFPExcept(FPExcept)':
2025-03-12T18:45:57.8554349Z /project/amrex/Src/Base/AMReX.cpp:979:5: error: 'fedisableexcept' was not declared in this scope; did you mean 'disableFPExcept'?
2025-03-12T18:45:57.8556262Z   979 |     fedisableexcept(flags);
2025-03-12T18:45:57.8556646Z       |     ^~~~~~~~~~~~~~~
2025-03-12T18:45:57.8556979Z       |     disableFPExcept
2025-03-12T18:45:57.8606731Z /project/amrex/Src/Base/AMReX.cpp: In function 'amrex::FPExcept amrex::enableFPExcept(FPExcept)':
2025-03-12T18:45:57.8608424Z /project/amrex/Src/Base/AMReX.cpp:994:5: error: 'feenableexcept' was not declared in this scope; did you mean 'feraiseexcept'?
2025-03-12T18:45:57.8610129Z   994 |     feenableexcept(flags);
2025-03-12T18:45:57.8610504Z       |     ^~~~~~~~~~~~~~
2025-03-12T18:45:57.8610825Z       |     feraiseexcept
```

Seen in https://github.com/BLAST-ImpactX/impactx-wheels

--- a/Src/Base/AMReX.cpp
+++ b/Src/Base/AMReX.cpp
@@ -608,7 +608,8 @@ amrex::Initialize (int& argc, char**& argv, bool build_parm_parse,
                 pp.queryAdd("fpe_trap_zero", divbyzero);
                 pp.queryAdd("fpe_trap_overflow", overflow);
 
-#if defined(__linux__)
+#if defined(__linux__) && defined(__GLIBC__)
+
                 curr_fpe_excepts = 0;
                 if (invalid)   { curr_fpe_excepts |= FE_INVALID;   }
                 if (divbyzero) { curr_fpe_excepts |= FE_DIVBYZERO; }

